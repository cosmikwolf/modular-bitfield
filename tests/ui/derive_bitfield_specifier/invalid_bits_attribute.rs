use modular_bitfield::prelude::*;

#[derive(Specifier)]
#[bits = 1]
#[bits = 1]
enum TooManyAttrs {
    Zero = 0,
    One = 1,
}

#[derive(Specifier)]
#[bits = 1.0]
enum NotAnInt {
    Zero = 0,
    One = 1,
}

fn main() {}
