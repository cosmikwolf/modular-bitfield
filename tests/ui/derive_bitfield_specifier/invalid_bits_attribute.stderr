error: More than one 'bits' attribute is not permitted
 --> tests/ui/derive_bitfield_specifier/invalid_bits_attribute.rs:5:1
  |
5 | #[bits = 1]
  | ^^^^^^^^^^^

error: bits must be integer literal for fixed size: #[bits = N]
  --> tests/ui/derive_bitfield_specifier/invalid_bits_attribute.rs:12:3
   |
12 | #[bits = 1.0]
   |   ^^^^^^^^^^
